### PR TITLE
Expose file inode number on linux and file index on windows

### PR DESCRIPTION
**Pull request by leroycep**
Adds `.inode` to `fs.File.Stat`, and a type named `fs.File.INode` which is a u64 on linux and an i64 (LARGE_INTEGER) on windows.

### CODE REVIEW

File: lib/std/fs/file.zig [hunk 1]
```diff
@@ -60,6 +64,10 @@ pub const File = struct {
         /// `error.FileAlreadyExists` to be returned.
         exclusive: bool = false,
 
+        /// Ensure that this open call creates the file, otherwise causes
```
Comment (andrewrk) [inline lib/std/fs/file.zig]: Looks like these doc comments need to be updated

Comment (leroycep) [reply lib/std/fs/file.zig]: Oh, actually, I need to remove that. I started working on adding locking, then decided to work on exposing the inode number instead. I'm planning on making a separate pull request for file locking

File: lib/std/fs/file.zig [hunk 2]
```diff
@@ -29,6 +29,15 @@ pub const File = struct {
 
     pub const Mode = os.mode_t;
 
+    /// The type that is used to represent the inode/file index number. On windows this is a
+    /// LARGE_INTEGER (i64), and on linux this is a u64.
+    pub const INode = switch (builtin.os.tag) {
+        .windows => os.windows.LARGE_INTEGER,
+        // TODO: Handle possibility of 128 bit numbers? ReFS on windows server 2012 uses a 128 bit file
+        // index. See https://docs.microsoft.com/en-us/windows/win32/api/fileapi/ns-fileapi-by_handle_file_information
+        else => u64,
```
Comment (andrewrk) [inline lib/std/fs/file.zig]: We actually do have a `std.os.ino_t` type, although the definition may be missing for some posix targets.

Comment (andrewrk) [reply lib/std/fs/file.zig]: I think in this case, even windows can define ino_t in lib/std/os/bits/windows.zig (similar to how fd_t is defined as HANDLE), and then we don't need a switch here, probably not even `INode`. The Stat struct could just use std.os.ino_t.

**Comment by andrewrk** [comment]
Nice work!

**Pull request merged**